src/Swashbuckle.AspNetCore.SwaggerGen/Generator/DocumentProvider.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System.IO;
+using System.Threading.Tasks;
+using Microsoft.Extensions.ApiDescription;
+using Microsoft.Extensions.Options;
+using Microsoft.OpenApi.Writers;
+using Swashbuckle.AspNetCore.Swagger;
+
+namespace Swashbuckle.AspNetCore.SwaggerGen
+{
+    internal class DocumentProvider : IDocumentProvider
+    {
+        private readonly SwaggerOptions _options;
+        private readonly ISwaggerProvider _swaggerProvider;
+
+        public DocumentProvider(IOptions<SwaggerOptions> options, ISwaggerProvider swaggerProvider)
+        {
+            _options = options.Value;
+            _swaggerProvider = swaggerProvider;
+        }
+
+        public Task GenerateAsync(string documentName, TextWriter writer)
+        {
+            // Let UnknownSwaggerDocument or other exception bubble up to caller.
+            var swagger = _swaggerProvider.GetSwagger(documentName, host: null, basePath: null);
+            var jsonWriter = new OpenApiJsonWriter(writer);
```

Comment:
Should this be disposed?

Reply:
No, the `OpenApiJsonWriter` is not disposable